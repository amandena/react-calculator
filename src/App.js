import './App.css'

function App() {
  return (
    <div className='App'>
      React Calculator
    </div>
  )
}

export default App
