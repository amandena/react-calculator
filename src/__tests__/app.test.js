import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import App from '../App'

afterEach(() => {
  cleanup()
})

test('should render App component', () => {
  render(<App/>)
  const appElement = screen.getByTestId('app-1')
  expect(appElement).toBeInTheDocument()
  expect(appElement).toHaveTextContent('DEL')
  expect(appElement).toContainHTML('</button>')
})

test('matches snapshot', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot()
})