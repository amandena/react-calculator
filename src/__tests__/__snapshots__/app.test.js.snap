// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="app-1"
>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      
       
      0
    </div>
    <div
      className="operators"
    >
      <button
        onClick={[Function]}
      >
        /
      </button>
      <button
        onClick={[Function]}
      >
        *
      </button>
      <button
        onClick={[Function]}
      >
        +
      </button>
      <button
        onClick={[Function]}
      >
        -
      </button>
      <button
        onClick={[Function]}
      >
        DEL
      </button>
    </div>
    <div
      className="digits"
    >
      <button
        onClick={[Function]}
      >
        1
      </button>
      <button
        onClick={[Function]}
      >
        2
      </button>
      <button
        onClick={[Function]}
      >
        3
      </button>
      <button
        onClick={[Function]}
      >
        4
      </button>
      <button
        onClick={[Function]}
      >
        5
      </button>
      <button
        onClick={[Function]}
      >
        6
      </button>
      <button
        onClick={[Function]}
      >
        7
      </button>
      <button
        onClick={[Function]}
      >
        8
      </button>
      <button
        onClick={[Function]}
      >
        9
      </button>
      <button
        onClick={[Function]}
      >
        0
      </button>
      <button
        onClick={[Function]}
      >
        .
      </button>
      <button
        onClick={[Function]}
      >
        =
      </button>
    </div>
  </div>
</div>
`;
